src/test/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,187 @@
+            new UpdateCartItemRequestDto()
+                    .setQuantity(5);
+    private static final AddCartItemRequestDto VALID_ADD_ITEM_REQUEST =
+            new AddCartItemRequestDto()
+                    .setBookId(VALID_ID)
+                    .setQuantity(5);
+
+    private static final CartItemDto CART_ITEM_DTO =
+            new CartItemDto()
+                    .setId(VALID_ID)
+                    .setTitle("Harry Potter")
+                    .setBookId(VALID_ID)
+                    .setQuantity(5);
+    private static final ShoppingCartDto SHOPPING_CART_DTO =
+            new ShoppingCartDto()
+                    .setId(VALID_ID)
+                    .setCartItems(Set.of(CART_ITEM_DTO))
+                    .setUserId(VALID_ID);
+    private static final Authentication AUTHENTICATION =
+            mock(Authentication.class);
+
+    static {
+        VALID_CART_ITEM.setShoppingCart(VALID_SHOPPING_CART);
+    }
+
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private CartItemRepository cartItemRepository;
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
+    @Mock
+    private CartItemMapper cartItemMapper;
+
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @Test
+    @DisplayName("Test retrieving an existing shopping cart for the user")
+    public void getShoppingCart_ExistingUser_ReturnsShoppingCart() {
+
+        when(AUTHENTICATION.getPrincipal()).thenReturn(VALID_USER);
+
+        when(shoppingCartRepository.findByUserId(VALID_USER.getId()))
+                .thenReturn(Optional.of(VALID_SHOPPING_CART));
+        when(shoppingCartMapper.toDto(VALID_SHOPPING_CART)).thenReturn(SHOPPING_CART_DTO);
+
+        ShoppingCartDto actual = shoppingCartService.getShoppingCart(AUTHENTICATION);
+
+        verify(shoppingCartRepository, times(1)).findByUserId(VALID_USER.getId());
+        verify(shoppingCartMapper, times(1)).toDto(VALID_SHOPPING_CART);
+
+        assertNotNull(actual);
+        assertEquals(SHOPPING_CART_DTO, actual);
+    }
+
+    @Test
+    @DisplayName("Test adding cart item to db")
+    public void addToCart_ValidRequest_AddsItemToShoppingCart() {
+        when(AUTHENTICATION.getPrincipal()).thenReturn(VALID_USER);
+        when(shoppingCartRepository.findByUserId(VALID_USER.getId()))
+                .thenReturn(Optional.of(VALID_SHOPPING_CART));
+        when(cartItemMapper.toEntity(VALID_ADD_ITEM_REQUEST)).thenReturn(VALID_CART_ITEM);
+        when(cartItemRepository.save(VALID_CART_ITEM)).thenReturn(VALID_CART_ITEM);
+        when(cartItemMapper.toDto(VALID_CART_ITEM)).thenReturn(CART_ITEM_DTO);
+
+        CartItemDto actual = shoppingCartService.addToCart(VALID_ADD_ITEM_REQUEST,
+                AUTHENTICATION);
+
+        assertNotNull(actual);
+        assertEquals(CART_ITEM_DTO, actual);
+
+        verify(AUTHENTICATION, times(1)).getPrincipal();
```

Comment:
Not necessary to verify things that you describe in when section, because if something that you define in when is not executed exception will be thrown

Reply:
Not resolved